libmamba/src/core/context.cpp

```diff
@@ -366,4 +366,23 @@ namespace mamba
         logger->dump_backtrace_no_guards();
     }
 
+    void Context::set_conda_build_local_paths(const std::vector<std::string>& vec)
+    {
+        conda_build_local_paths = vec;
+    }
+
+    std::vector<std::string> Context::get_conda_build_local_paths() const
+    {
+        if (conda_build_local_paths.empty())
+        {
+            return { prefix_params.target_prefix.string() + "/conda-bld",
```

Comment:
I don't know if it has been intentional, but this differs from what `conda.base.context.conda_build_local_paths` would have:
https://github.com/conda/conda/blob/23.11.0/conda/base/context.py#L505-L523
1. This adds non-existent folders whereas `conda` only adds existing ones.
2. `bld_path` config value (and as such `${CONDA_BLD_PATH}`) is not used.
3. `${CONDA_PREFIX}/conda-bld` is added whereas `conda` would never add this
    (unless it's the root prefix which always considered).
4. `conda_build: root-dir` config values is not used.

TBH, I didn't even know that `conda_build: root-dir` existed -- and also not that `conda-build` uses that config entry but not `bld_path` (it only checks the environment variable):
https://github.com/conda/conda-build/blob/3.28.2/conda_build/config.py#L467-L468

----
So, yeah, the behavior in `conda`/`conda-build` is a bit convoluted..
But at least the `${CONDA_PREFIX}/conda-bld` addition and `${CONDA_BLD_PATH}` omission here seem rather unexpected to me.

Reply:
This keeps the existing paths in mamba with the option to change this from the API. In `conda-libmamba-solver`, we set these values from conda's context, so they will be optional. So, if mamba is used from `conda-libmamba-solver`, the behaviour will be the same. I agree that it differs when using mamba directly.

Reply:
Ah, didn't see @jaimergp's comment in https://github.com/mamba-org/mamba/pull/3056#issuecomment-1851576025 which iterates the same points.
So, why do we want different behaviors between the two implementations?

Reply:
> This keeps the existing paths in mamba with the option to change this from the API. In `conda-libmamba-solver`, we set these values from conda's context, so they will be optional. So, if mamba is used from `conda-libmamba-solver`, the behaviour will be the same. I agree that it differs when using mamba directly.

I'd consider this behavior a bug in Mamba and not worth preserving...

Reply:
I guess in case we assume that we only use this via `conda-libmamba-solver` with the value overridden, then it's probably just a thing to change for the `2.x` versions in `main`.

Reply:
Added https://github.com/mamba-org/mamba/issues/3124 to track this.